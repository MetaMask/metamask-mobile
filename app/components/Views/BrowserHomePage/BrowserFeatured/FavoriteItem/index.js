import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet, View } from 'react-native';
import WebsiteIcon from '../../../../UI/WebsiteIcon';
import { colors, fontStyles } from '../../../../../styles/common';

const styles = StyleSheet.create({
	wrapper: {
		padding: 18,
		flex: 1,
		flexDirection: 'row'
	},
	icon: {
		height: 35,
		width: 35
	},
	iconWrapper: {
		borderWidth: 2,
		borderRadius: 10,
		height: 88,
		width: 88,
		borderColor: colors.lightGray,
		alignItems: 'center',
		justifyContent: 'center',
		shadowOpacity: 0.2
	},
	contentWrapper: {
		paddingHorizontal: 18
	},
	name: {
		fontSize: 16,
		marginBottom: 7,
		...fontStyles.bold
	},
	description: {
		...fontStyles.normal
	}
});

/**
 * Favorites
 */
export default class FavoriteItem extends Component {
	static propTypes = {
		/**
		 * Dapp description
		 */
		description: PropTypes.string,
		/**
		 * Dapp name
		 */
		name: PropTypes.string,
		/**
		 * Dapp url
		 */
		url: PropTypes.string
	};

	render() {
		const { name, url, description } = this.props;
		return (
			<View style={styles.wrapper}>
				<View style={styles.iconWrapper} elevation={5}>
					<WebsiteIcon style={styles.icon} title={name} url={url} />
				</View>
				<View style={styles.contentWrapper}>
					<Text style={styles.name}>{name}</Text>
					<Text style={styles.description}>{description}</Text>
				</View>
			</View>
		);
	}
}
