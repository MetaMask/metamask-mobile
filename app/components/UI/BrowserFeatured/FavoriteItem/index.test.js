import React from 'react';
import { shallow } from 'enzyme';
import FavoriteItem from './';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

describe('FavoriteItem', () => {
	it('should render correctly', () => {
		const initialState = {};
		const wrapper = shallow(<FavoriteItem name={'name'} url={'url'} description={'description'} />, {
			context: { store: mockStore(initialState) }
		});
		expect(wrapper.dive()).toMatchSnapshot();
	});
});
