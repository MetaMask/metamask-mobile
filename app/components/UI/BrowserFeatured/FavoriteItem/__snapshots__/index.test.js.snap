// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteItem should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
